### ISSUE TITLE
an 0.18 patch for 0.7?

### ISSUE THREAD

**Issue by ExpandingMan** [JuliaOpt/JuMP.jl#1196]
What's the current thinking on this?  Is this already planned?  To be honest, I'm not sure my knowledge of git is even good enough to make a PR to do this.  It would be nice to have a version of 0.18 that works nicely on 0.7 to help ease the transition.

**Comment by mlubin** [comment]
> What's the current thinking on this? Is this already planned?

We want to release a version of JuMP 0.18 (e.g., 0.18.1) that runs on both 0.6 and 0.7. However, I'm not aware of any active work in this direction.

> I'm not sure my knowledge of git is even good enough to make a PR to do this.

It's pretty simple. Check out `release-0.18` (`Pkg.checkout` or `git checkout`) and make changes as you normally would. When you go to make a PR on github, you can select to make it for the `release-0.18` branch instead of `master`.

**Comment by ExpandingMan** [comment]
Thanks!

So, my understanding is, what is on JuMP master right now is for MathOptInterface, so I should not generally expect to use code past the 0.18 tag.  Instead, I should branch from the `release-0.18`.  I'll also have to update MathProgInterface which I can do on its master.  Correct?

I'm definitely interested in helping with this if I have time.  What will things be like with the individual solvers?  It looks like work has been done to get them going on 0.7, but have many of them been changed to MatOptInterface already?

Sorry, this probably isn't the appropriate place to be discussing this.  Will get on gitter...

**Comment by blegat** [comment]
> have many of them been changed to MatOptInterface already?

It has been done on separate package, e.g. MathOptInterfaceGurobi.jl so yes you can push on master directly for solvers.

**Comment by mlubin** [comment]
`master` uses MOI. `release-0.18` uses MPB. You don't need to install MOI to use `release-0.18`.

> What will things be like with the individual solvers? 

They also need compatibility updates for 0.7. Some 0.7 issues have been addressed already, but not all.

**Comment by ExpandingMan** [comment]
Looking at some of these it seems as if they are mainly using LinQuadOptInterface.  What is that, is that going to be added to MathOptInterface at some point?

**Comment by blegat** [comment]
> Looking at some of these it seems as if they are mainly using LinQuadOptInterface. What is that, is that going to be added to MathOptInterface at some point?

I don't think so, LinQuadOptInterface (LQOI) is useful for LP solvers to help them implement MathOptInterface but conic and non-linear solvers don't use LQOI
